Sources/TuistDependencies/DependenciesManagerInteractors/Interactors/CarthageFrameworksInteractor.swift

```diff
@@ -0,0 +1,121 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistSupport
+
+/// Saving frameworks installed using `carthage`.
+public protocol CarthageFrameworksInteracting {
+    /// Saves frameworks installed using `carthage`.
+    /// - Parameters:
+    ///   - path: Directory whose project's dependencies will be installed.
+    ///   - temporaryDirectoryPath: Folder where dependencies are being installed.
+    func save(at path: AbsolutePath, temporaryDirectoryPath: AbsolutePath) throws
+}
+
+public final class CarthageFrameworksInteractor: CarthageFrameworksInteracting {
+    private let fileHandler: FileHandling
+
+    public init(fileHandler: FileHandling = FileHandler.shared) {
+        self.fileHandler = fileHandler
+    }
+
+    public func save(at path: AbsolutePath, temporaryDirectoryPath: AbsolutePath) throws {
+        let buildDirectoryPath = temporaryDirectoryPath.appending(components: "Carthage", "Build")
+        let dependenciesDirectoryPath = path.appending(components: Constants.tuistDirectoryName, Constants.DependenciesDirectory.name)
+
+        let versionFiles = try readVersionFiles(at: buildDirectoryPath)
+
+        var alreadyCopiediOSFramworks = Set<String>()
```

Comment:
I'm not sure I fully understand why something like this is necessary. Is it possible to have a Carthage Build directory with the same framework more than once? When would that scenario happen?